projects/igniteui-angular/src/lib/combo/combo-dropdown.component.ts

```diff
@@ -205,127 +205,107 @@ export class IgxComboDropDownComponent extends IgxDropDownBase implements OnDest
+            }
         }
     }
 
-    private navigateVirtualItem(direction: Navigate, extraScroll?: number) {
-        const vContainer = this.verticalScrollContainer;
-        // If the data is vitualized, data.length === vContainer.chunkSize, so the below checks are no-longer valid
-        if (vContainer && vContainer.totalItemCount && vContainer.totalItemCount !== 0) {
-            this.navigateRemoteItem(direction);
-            return;
-        }
-        let state = vContainer.state;
-        if (this.isScrolledToLast && direction === Navigate.Down) { // If on the bottom most item, do not subscribe
-            return;
-        }
-        // If on the topmost item, do not subscribe
-        if (this.verticalScrollContainer.getVerticalScroll().scrollTop === 0 && direction === Navigate.Up) {
+    private navigateDown(allData: any[], vContainer: IgxForOfDirective<any>, extraScroll: number, newIndex?: number) {
+        const focusedItem = this.focusedItem;
+        const items = this.items;
+        const children = this.children.toArray();
+        if (focusedItem && (focusedItem.value === allData[allData.length - 1] || focusedItem.value === 'ADD ITEM')) { // If very last item
+            if (this.combo.isAddButtonVisible() && focusedItem.value !== 'ADD ITEM') { // If add button is visible
+                this.focusItem(items.length - 1); // Focus add button
+            }
             return;
         }
-        const isScrollUp = direction === Navigate.Up;
-        let newScrollStartIndex = isScrollUp ? state.startIndex - 1 : state.startIndex + 1;
-        if (newScrollStartIndex < 0) {
-            newScrollStartIndex = 0;
+        let targetDataIndex = newIndex === -1 ? this.itemIndexInData(this.focusedItem.index) + 1 : this.itemIndexInData(newIndex);
+        const lastLoadedIndex = vContainer.state.startIndex + vContainer.state.chunkSize - 1; // Last item is not visible, so require scroll
+        if (targetDataIndex < lastLoadedIndex) { // If no scroll is required
+            if (newIndex !== -1 || newIndex === children.length - 1 - extraScroll) { // Use normal nav for visible items
+                super.navigateItem(newIndex);
+            }
+        } else if (this.isScrolledToLast && targetDataIndex === lastLoadedIndex) { // If already at bottom and target is last item
+            this.focusItem(items.length - 1 - extraScroll); // Focus the last item (excluding Add Button)
+        } else { // If scroll is required
+            // If item is header, find next non-header index
+            const addedIndex = allData[targetDataIndex].isHeader ?
+            [...allData].splice(targetDataIndex, allData.length - 1).findIndex(e => !e.isHeader) : 0;
+            targetDataIndex += addedIndex; // Add steps to the target index
+            if (addedIndex === -1 && this.combo.isAddButtonVisible()) { // If there are no more non-header items & add button is visible
+                this.focusItem(items.length);
```

Comment:
I'm trying to figure out if focusItem in the combo-drop-down, is not almost the same as navigateItem in the parent drop-down. If so, this means that instead of this.focusItem we can call super.navigateItem and simplify the code. If not, please ignore that.